pilot_data_analysis.Rmd

```diff
@@ -226,12 +226,17 @@ Create a dataframe to identify the first look and see if the first look is >100m
  
 # gal's approach
 df_t_first_aoi <- d %>% 
+  # for now only look at exp1, and at window between bear disappearing and point of disambiguation (+120 to account for time to react)
   filter(., experiment_num == "pilot_1a",
-         t >= -4000, t <= 120, 
-         aoi %in% c("target", "distractor")) %>% 
-        group_by(subid, trial_num) %>% 
-  mutate(first_look_aoi = rle(aoi)$values[rle(aoi)$lengths > 4][1]) %>% # 4 consecutive timepoints looking at target or distractor is counted as first look
-    ungroup() 
+         t >= -4000+120, t <= 120) %>% 
+  # create dummy variable for AOI in order to exploit rle, which handles NaN's neatly
+   mutate(aoi_dummy = case_when(aoi == "other" ~ NaN, aoi == "target" ~ 1, aoi == "distractor" ~ 0)) %>%
```

Comment:
you can use NA in a character vector right, and this will work with rle()?

like `mutate(aoi = ifelse(aoi == "other", NA, aoi)) %>%`